sws_context_test.go

```diff
@@ -0,0 +1,66 @@
+package astiav_test
+
+import (
+	"image"
+	"reflect"
+	"testing"
+
+	"github.com/asticode/go-astiav"
+	"github.com/stretchr/testify/require"
+)
+
+// Test constants for source and destination dimensions and formats
+const (
+	srcW = 100
+	srcH = 100
+	dstW = 200
+	dstH = 200
+)
+
+// assertImageType is a helper function to check the type of an image.
+func assertImageType(t *testing.T, img image.Image, expectedType reflect.Type) {
+	actualType := reflect.TypeOf(img)
+	require.Equal(t, expectedType, actualType, "Image type does not match")
+}
+
+// TestSWS tests the scaling functionality provided by the SWSContext.
+func TestSWS(t *testing.T) {
+	// Allocate and initialize source and destination frames
+	srcFrame := astiav.AllocFrame()
+	defer srcFrame.Free()
+	dstFrame := astiav.AllocFrame()
+	defer dstFrame.Free()
+
+	srcFrame.SetHeight(srcH)
+	srcFrame.SetWidth(srcW)
+	srcFrame.SetPixelFormat(astiav.PixelFormatYuv420P)
+	srcFrame.AllocBuffer(1)
+
```

Comment:
i adding PrepareDestinationFrameForScaling i find that we should have a short func for these because if you setup scale context with all dst attributes we already know what user want for the scaling.

```go
func (ssc *SoftwareScaleContext) PrepareDestinationFrameForScaling(dstFrame *Frame) {
	dstFrame.SetPixelFormat(PixelFormat(ssc.dstFormat))
	dstFrame.SetWidth(int(ssc.dstW))
	dstFrame.SetHeight(int(ssc.dstH))
	dstFrame.AllocBuffer(1)
}
```